docs/overview/components.rst

```diff
@@ -0,0 +1,14 @@
+Components
```

Comment:
I know this is a WIP, just leaving this as a reminder to flesh out this section more.